src/Utilities/Compiler/Extensions/ITypeSymbolExtensions.cs

```diff
@@ -287,6 +287,57 @@ public static Accessibility DetermineMinimalAccessibility(this ITypeSymbol typeS
             return typeSymbol.Accept(MinimalAccessibilityVisitor.Instance);
         }
 
+        public static bool HasAnyCollectionCountProperty(this ITypeSymbol invocationTarget, WellKnownTypeProvider wellKnownTypeProvider)
```

Comment:
I am not sure whether this should take:

1/ A `Compilation` object.

2/ The `WellKnownTypeProvider`.

3/ The 3 `INamedTypeSymbol`.

Reply:
What you have currently is fine.